test/unit/bypass_validation_tests.js

```diff
@@ -0,0 +1,469 @@
+'use strict';
+
+const MongoClient = require('../..').MongoClient;
+const expect = require('chai').expect;
+const mock = require('mongodb-mock-server');
+
+describe('bypass document validation', function() {
+  const test = {};
+  beforeEach(() => {
+    return mock.createServer().then(server => {
+      test.server = server;
+    });
+  });
+  afterEach(() => mock.cleanup());
+
+  // aggregate
+  it('should only set bypass document validation if strictly true in aggregate', function(done) {
```

Comment:
A lot of the code in this file is repeated. Could we refactor it to be more DRY?